### PR TITLE
polar: init at 1.3.0

### PR DESCRIPTION
**Pull request by aureleoules**
###### Motivation for this change
Adds Polar v1.3.0
> One-click Bitcoin Lightning networks for local app development & testing

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/polar/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  installPhase = ''
+    mv usr $out
+    mv opt $out
+    substituteInPlace $out/share/applications/${pname}.desktop --replace /usr/ $out/
+  '';
+
+  meta = with lib; {
+    description =
+      "One-click Bitcoin Lightning networks for local app development & testing";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/polar/default.nix:69]:
```suggestion
    description = "One-click Bitcoin Lightning networks for local app development & testing";
```

File: pkgs/applications/blockchains/polar/default.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+, at-spi2-core, mesa, libxkbcommon }:
+
+stdenv.mkDerivation rec {
+  pname = "polar";
+  version = "1.3.0";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url =
+      "https://github.com/jamaljsr/polar/releases/download/v${version}/${pname}-linux-amd64-v${version}.deb";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/polar/default.nix:14]:
```suggestion
    url = "https://github.com/jamaljsr/polar/releases/download/v${version}/${pname}-linux-amd64-v${version}.deb";
```

File: pkgs/applications/blockchains/polar/default.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
+{ fetchurl, lib, stdenv, autoPatchelfHook, dpkg, curl, libX11, libXext, squashfsTools
+, xorg, makeWrapper, wrapGAppsHook, freetype, glib, pango
+, atk, gdk-pixbuf, gtk3, cups, nspr, nss, libpng, libnotify, libgcrypt
+, systemd, fontconfig, dbus, ffmpeg, at-spi2-atk
+, at-spi2-core, mesa, libxkbcommon }:
+
+stdenv.mkDerivation rec {
+  pname = "polar";
+  version = "1.3.0";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/polar/default.nix:10]:
```suggestion
```
this is done automatically

**Comment by aureleoules** [comment]
Can this PR be merged?

File: pkgs/applications/blockchains/polar/default.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/jamaljsr/polar/releases/download/v${version}/${pname}-linux-amd64-v${version}.deb";
+    hash = "sha256:0wq40qf33wmyfbf5xrlqjpamfgc71nrfdkq7yn4ai0vk0vpbjcj3";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook dpkg ];
+
+  buildInputs = [
+      curl
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/polar/default.nix:19]:
```suggestion
    curl
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
You are missing a top-level entry.

**Comment by aureleoules** [comment]
> You are missing a top-level entry.

Which one?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Missing top-level entry

File: pkgs/applications/blockchains/polar/default.nix [hunk 5]
```diff
@@ -0,0 +1,84 @@
+{ fetchurl, lib, stdenv, autoPatchelfHook, dpkg, curl, libX11, libXext, squashfsTools
+, xorg, makeWrapper, wrapGAppsHook, freetype, glib, pango
+, atk, gdk-pixbuf, gtk3, cups, nss, libpng, libnotify, libgcrypt
+, systemd, fontconfig, ffmpeg, dbus, at-spi2-atk
+, at-spi2-core, mesa, libxkbcommon, makeDesktopItem }:
+
+let
+  desktopItem = makeDesktopItem {
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/polar/default.nix:8]: That is unused. I think you want to use copyDesktopItems.

**Pull request closed without merging**